### ISSUE TITLE
std::process::Command::output is unacceptably slow

### ISSUE THREAD

**Issue by safinaskar** [rust-lang/rust#108223]
`std::process::Command::output` is unacceptably slow. Consider this code:

https://play.rust-lang.org/?version=nightly&mode=release&edition=2021&gist=333b57442699553257f0273bb1bba3ac

When trying this code, make sure you have enabled release mode. On my local machine I see this output:
```text
1565 ms
100000000
103 ms
100000000
```

So `std::process::Command::output` is 15x slower than `std::io::BufReader::read_to_end()`. My rustc version is `rustc 1.69.0-nightly (75a0be98f 2023-02-05)`. My OS is Debian Stretch x86_64

**Comment by Nilstrieb** [comment]
On windows they are both equally fast, so it seems to be a unix-specific thing. Also note that these two programs are not equivalent, as `output()` also reads `stderr`, while you don't (this probably introduces some overhead). Still seems worth at least investigating.

**Comment by KittyBorgX** [comment]
What's the magnitude of slowness here? and @Nilstrieb what's the time taken on windows?
Here's it on my computer, built with release mode 
```
614 ms
100000000
36 ms
100000000
```

My computer : Apple M1 Mac Mini, 16gb RAM and 8 CPU Cores

**Comment by safinaskar** [comment]
@tmiasko . I just compared old rustc (`rustc 1.69.0-nightly (e972bc808 2023-01-29)`) and new one, with this fix (`rustc 1.69.0-nightly (c5c7d2b37 2023-02-24)`). I see that the bug is actually fixed, thanks a lot! Moreover, both benchmarks became faster
